novawallet/Modules/Swaps/Setup/SwapSetupInteractor.swift

```diff
@@ -1,7 +1,181 @@
+        }
+
+        priceProviders[chainAsset.chainAssetId] = subscribeToPrice(
+            for: priceId,
+            currency: currencyManager.selectedCurrency
+        )
+    }
+
+    private func quote(args: AssetConversion.QuoteArgs) {
+        clear(cancellable: &quoteCall)
+
+        let wrapper = assetConversionOperationFactory.quote(for: args)
+        wrapper.targetOperation.completionBlock = { [weak self] in
+            DispatchQueue.main.async {
+                guard self?.quoteCall === wrapper else {
+                    return
+                }
+                do {
+                    let result = try wrapper.targetOperation.extractNoCancellableResultData()
+
+                    self?.presenter?.didReceive(quote: result)
+                } catch {
+                    self?.presenter?.didReceive(error: .quote(error))
+                }
+            }
+        }
+
+        quoteCall = wrapper
+        operationQueue.addOperations(wrapper.allOperations, waitUntilFinished: false)
+    }
+
+    private func update(chainModel: ChainModel) {
+        guard let metaChainAccountResponse = selectedAccount.fetchMetaChainAccount(
+            for: chainModel.accountRequest()
+        ) else {
+            return
+        }
+        extrinsicService = extrinsicServiceFactory.createService(
+            account: metaChainAccountResponse.chainAccount,
+            chain: chainModel
+        )
+        accountId = metaChainAccountResponse.chainAccount.accountId
+    }
+
+    private func fee(args: AssetConversion.CallArgs) {
+        clear(cancellable: &runtimeOperationCall)
+        guard let extrinsicService = extrinsicService else {
+            return
+        }
+
+        let runtimeCoderFactoryOperation = runtimeService.fetchCoderFactoryOperation()
+
+        runtimeCoderFactoryOperation.completionBlock = { [weak self] in
+            guard let self = self else {
+                return
+            }
+            do {
+                let runtimeCoderFactory = try runtimeCoderFactoryOperation.extractNoCancellableResultData()
+                let builder = self.assetConversionExtrinsicService.fetchExtrinsicBuilderClosure(
+                    for: args,
+                    codingFactory: runtimeCoderFactory
+                )
+                self.feeProxy.estimateFee(using: extrinsicService, reuseIdentifier: "", setupBy: builder)
+            } catch {
+                DispatchQueue.main.async {
+                    self.presenter?.didReceive(error: .fetchFeeFailed(error))
+                }
+            }
+        }
+
+        runtimeOperationCall = runtimeCoderFactoryOperation
+        operationQueue.addOperation(runtimeCoderFactoryOperation)
+    }
+}
+
+extension SwapSetupInteractor: SwapSetupInteractorInputProtocol {
+    func setup() {
+        feeProxy.delegate = self
+    }
+
+    func calculateQuote(for args: AssetConversion.QuoteArgs) {
+        quote(args: args)
+    }
+
+    func set(chainModel: ChainModel) {
```

Comment:
I think we also need an interface for updating assets. Because we need to subscribe to balances for those assets. Probably, we can replace this for 2 methods: setPayAsset and setReceiveAsset. In this case we will use chain from setPayAsset to create Extrinsic Service.